COMETwebapp/ViewModels/Components/ReferenceData/CategoriesTableViewModel.cs

```diff
@@ -217,5 +233,134 @@ private void UpdateRows(IEnumerable<CategoryRowViewModel> rowsToDisplay)
+            clonedCateory.IsDeprecated = false;
+            cateoryToUnDeprecate.Add(clonedCateory);
+            try
+            {
+                await this.sessionService.UpdateThingsSiteDirectory(cateoryToUnDeprecate);
+            }
+            catch (Exception exception)
+            {
+                Console.WriteLine(exception.Message);
+                throw;
+            }
+            this.popupVisible = false;
+        }
+
+        /// <summary>
+        ///     Tries to deprecate a <see cref="Category" />
+        /// </summary>
+        /// <returns>A <see cref="Task" /></returns>
+        public async Task DeprecatingCategory()
+        {
+            var categoryToDeprecate = new List<Category>();
+            var clonedCategory = this.Category.Clone(false);
+            clonedCategory.IsDeprecated = true;
+            categoryToDeprecate.Add(clonedCategory);
+            try
+            {
+                await this.sessionService.UpdateThingsSiteDirectory(categoryToDeprecate);
+            }
+            catch (Exception exception)
+            {
+                Console.WriteLine(exception.Message);
+                throw;
+            }
+            this.popupVisible = false;
+        }
+
+        /// <summary>
+        ///     Tries to create a new <see cref="Category" />
+        /// </summary>
+        /// <returns>A <see cref="Task" /></returns>
+        public async Task AddingCategory()
+        {
+            var thingsToCreate = new List<Thing>();
+
+            if (this.SelectedSuperCategories.Any())
+            {
+                this.Category.SuperCategory = this.SelectedSuperCategories.ToList();
+            }
+            if (this.SelectedPermissibleClasses.Any())
+            {
+                this.Category.PermissibleClass = this.SelectedPermissibleClasses.Select(x => x.ClassKind).ToList();
+            }
+            this.Category.Container = this.SelectedReferenceDataLibrary;
+            thingsToCreate.Add(this.Category);
+            var clonedRDL = this.SelectedReferenceDataLibrary.Clone(false);
+            clonedRDL.DefinedCategory.Add(this.Category);
+            try
+            {
+                await this.sessionService.CreateThings(clonedRDL, thingsToCreate);
+            }
+            catch (Exception exception)
+            {
+                Console.WriteLine(exception.Message);
+                throw;
+            }
+        }
+
+        /// <summary>
+        ///     Method invoked when the component is ready to start, having received its
+        ///     initial parameters from its parent in the render tree.
+        ///     Override this method if you will perform an asynchronous operation and
+        ///     want the component to refresh when that operation is completed.
+        /// </summary>
+        /// <returns>A <see cref="Task" /> representing any asynchronous operation.</returns>
+        public void OnInitializedAsync()
```

Comment:
this method is not Async. Remove the Async on the method name and remove the return on documentation